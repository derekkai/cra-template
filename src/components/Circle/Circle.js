import PropTypes from 'prop-types'
import useStyles from 'hooks/useStyles'
import styles from './Circle.module.scss'

const Circle = ({ percent = 0 }) => {
  useStyles(styles)
  return (
    <svg className={styles.root} viewBox="0 0 200 200">
      <path
        style={{
          strokeDashoffset: (471 / 100) * (100 - percent),
        }}
        d=" M 100, 100 m -75, 0 a 75,75 0 1,0 150,0 a 75,75 0 1,0 -150,0 "
      />
    </svg>
  )
}

Circle.propTypes = {
  percent: PropTypes.number.isRequired,
}

export default Circle
