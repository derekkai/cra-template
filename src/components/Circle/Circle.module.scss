.root {
  path {
    stroke-linecap: round;
    stroke-linejoin: round;
    stroke: navy;
    stroke-width: 5px;
    fill: none;
    stroke-dasharray: 471;
    transition: all 0.2s linear;
  }

  &:hover {
    cursor: pointer;
    path {
      stroke-dashoffset: 0;
    }
  }
}
