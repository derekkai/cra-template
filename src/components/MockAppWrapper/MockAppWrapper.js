import PropTypes from 'prop-types'
import App from 'App'
import { store } from 'store/configureStore'

const initialState = {}

const MockAppWrapper = ({ children }) => (
  <App store={store} insertCss={() => {}}>
    {children}
  </App>
)

MockAppWrapper.propTypes = {
  children: PropTypes.node.isRequired,
}

export default MockAppWrapper
