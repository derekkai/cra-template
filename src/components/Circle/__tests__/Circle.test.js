import React from 'react'
import renderer from 'react-test-renderer'
import MockAppWrapper from 'components/MockAppWrapper/MockAppWrapper'
import ReactDOM from 'react-dom'
import Circle from '../Circle'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <MockAppWrapper>
      <Circle percent={0} />
    </MockAppWrapper>,
    div,
  )
})

it('render correctly', () => {
  const wrapper = renderer
    .create(
      <MockAppWrapper>
        <Circle percent={49} />
      </MockAppWrapper>,
    )
    .toJSON()

  expect(wrapper).toMatchSnapshot()
})
