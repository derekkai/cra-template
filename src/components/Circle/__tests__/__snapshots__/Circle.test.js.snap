// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<svg
  className="root"
  viewBox="0 0 200 200"
>
  <path
    d=" M 100, 100 m -75, 0 a 75,75 0 1,0 150,0 a 75,75 0 1,0 -150,0 "
    style={
      Object {
        "strokeDashoffset": 240.21,
      }
    }
  />
</svg>
`;
